### PR TITLE
Feat/add group chat

### PR DESCRIPTION
**Pull request by rudenkoserhii**
## Describe your changes

- I added popup windows for creating group chat

Screenshot 2024-01-01 201111
Screenshot 2024-01-01 201154

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://testhomework4.atlassian.net/browse/AG-218?atlOrigin=eyJpIjoiMTk0MThlY2I3ODdkNDE2MzhjNmFkOTUzODhiOTEwMWQiLCJwIjoiaiJ9)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/sections/chat/components/create-group-chat/formFirst.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
+'use client';
+
+import { useForm } from 'react-hook-form';
+import { Button, TextField, Typography, Box, Stack } from '@mui/material';
+import { patterns } from 'constants/patterns';
+
+// type TFunction = (key: string) => string;
+
```
Comment (AKutusenko) [inline src/sections/chat/components/create-group-chat/formFirst.tsx]: Commented code

File: src/sections/chat/components/create-group-chat/formSecond.tsx [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+}: Props): JSX.Element {
+  const [options, setOptions] = useState<Options>([]);
+  const initialOptions = options.filter((option) =>
+    initialMembers.some((member) => member.id === option.id)
+  );
+
+  const [value, setValue] = useState<Option | null>(initialOptions[0] || null);
+
+  const [members, setMembers] = useState<Members>(initialMembers);
+  // const [value, setValue] = useState<Option | null>();
```
Comment (AKutusenko) [inline src/sections/chat/components/create-group-chat/formSecond.tsx]: Comment

File: src/sections/chat/components/create-group-chat/formSecond.tsx [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+        disabled={options.length === 0}
+        ListboxProps={{ style: { maxHeight: '10rem' } }}
+        getOptionKey={(option) => option.id}
+        isOptionEqualToValue={(option, optionValue) => option.valueOf === optionValue.valueOf}
+        renderInput={(params) => (
+          <TextField
+            autoFocus
+            {...params}
+            label={options.length > 0 ? 'Enter agent name' : 'No more agents'}
+          />
```
Comment (AKutusenko) [inline src/sections/chat/components/create-group-chat/formSecond.tsx:122]: Text in the code

**Review by AKutusenko — APPROVED** [review]

File: src/sections/chat/components/create-group-chat/formFirst.tsx (hunk 1 above)
Comment (rudenkoserhii) [reply src/sections/chat/components/create-group-chat/formFirst.tsx]: fixed

File: src/sections/chat/components/create-group-chat/formSecond.tsx (hunk 2 above)
Comment (rudenkoserhii) [reply src/sections/chat/components/create-group-chat/formSecond.tsx]: fixed

File: src/sections/chat/components/create-group-chat/formSecond.tsx (hunk 3 above)
Comment (rudenkoserhii) [reply src/sections/chat/components/create-group-chat/formSecond.tsx:122]: fixed

**Pull request merged**